compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -298,6 +299,10 @@ class TreePickler(pickler: TastyPickler) {
 
   def pickleMethodic(tag: Int, tpe: LambdaType, mods: FlagSet)(using Context): Unit = {
     writeByte(tag)
+    if tag == ERASEDMETHODtype then
+      withLength {
+        tpe.erasedParams.foreach { isErased => writeByte(if isErased then 1 else 0) }
```

Comment:
We should use one byte for 8 parameters. I belive we have some other logic doing something similar.

Reply:
If we pickle this information after the flags
```scala
      if (mods != EmptyFlags) pickleFlags(mods, tpe.isTermLambda)
      tpe.erasedParams.foreach { isErased => writeByte(if isErased then 1 else 0) }
```
we might be able to read them without the need for `ERASEDMETHODtype`. After reading the mods we can check if the `Erased` flag is set and if it is we read the param erasure information.

Reply:
An alternative would be to use an `AnnotatedType` to tag the erase parameter types. Then, when we unpickle, we can check if it is annotated to know if the parameter should be erased or not. We would also strip those type annotations. Effectively pickling `def f(erased x: T): R` as `def f(x: T @erasedParam): R`.